[docs] Add Cookbook entry for Verilog case equality (IsX) (#4605)
chipsalliance/chisel: docs/src/cookbooks/cookbook.md

compile(new TooWideOrNarrowUInt(8, 4))
```

## How can I use Verilog "case equality" operators in Chisel?

Verilog has "case equality" (`===`) and inequality (`!==`) operators.
They are typically used to ignore unknown (`X`) values in assertions.

> Chisel does not support SystemVerilog `X` directly, but it is possible to check if a value is `X` with `chisel3.util.circt.isX`.

Reviewer: Mega-nit, but `X` is referred to as a Verilog feature above, so this should probably be consistent with that.

```suggestion
Chisel does not support Verilog `X` directly, but it is possible to check if a value is `X` with `chisel3.util.circt.isX`.
```

Author: Agreed, thanks!

`isX` is commonly used to guard assertions against `X` which gives similar behavior to Verilog case equality.

```scala mdoc:silent:reset
import chisel3._
import chisel3.util.circt.IsX

> class XSafeAssert extends Module {

Reviewer 2: ```suggestion
class AssertButAllowX extends Module {
```

perhaps?

Author: Sure!

  val in = IO(Input(UInt(8.W)))

>   // Assert that in is never zero, but also guard against X

Reviewer 2: this comment is a little misleading, we aren't "guarding against X" we are "do not trigger assert in the presence of X"

>   // Assert that in is never zero, but also guard against X

Reviewer 2: ```suggestion
  // Assert that `in` is never zero, but also allow X
```

Author: I phrased it differently, what do you think?

>   // Assert that in is never zero, but also guard against X

Author: ```suggestion
  // Assert that in is never zero; also do not trigger assert in the presence of X.
```

  assert(IsX(in) || in =/= 0.U, "in should never equal 0")
}
```

```scala mdoc:invisible
// Hidden but will make sure this actually compiles

> chisel3.docs.emitSystemVerilog(new XSafeAssert)

Author: ```suggestion
chisel3.docs.emitSystemVerilog(new AssertButAllowX)
```
